### PR TITLE
Added decorateWithHierarchySpecTag(..) method to be used to create 

### PR DESCRIPTION
**Pull request by iuliana**
the hierarchical spec tag when deploying blueprints.

### CODE REVIEW

**Review by duncangrant — APPROVED** [review]

File: core/src/main/java/org/apache/brooklyn/core/typereg/RegisteredTypes.java [hunk 1]
```diff
@@ -333,7 +332,20 @@ public static RegisteredType addAliases(RegisteredType type, Iterable<String> al
     @Beta
     public static RegisteredType addTag(RegisteredType type, Object tag) {
         if (tag!=null) {
-            ((BasicRegisteredType)type).tags.add( tag );
+            if (tag instanceof Map &&( (Map<String,Object>) tag).containsKey(BrooklynTags.YAML_SPEC_HIERARCHY)) {
```
Comment (ahgittin) [inline core/src/main/java/org/apache/brooklyn/core/typereg/RegisteredTypes.java:335]: special-casing this tag here feels wrong.  feels like the logic belongs in `BrooklynTags` or `HierarchySpecTag`.

**Comment by ahgittin** [comment]
nice work.  minor comment can be dealt with later.

not 100% sure we have all the data we want for context but we can iterate from here.  there is one a closely related feature i've wanted which is to indicate on a spec which ancestor brought it in; this could be handled by a `level_in_parent_yaml` field in your hierarchy spec tag.  might add that.

**Comment by ahgittin** [comment]
i get two related test failures though:

```
Failed tests: 
  EntityResourceTest.testGetSpecAcceptingXYaml:93->testGetSpec:126 expected [name: ent-with-spec
services:
- type: org.apache.brooklyn.entity.stock.BasicEntity] but found []
  EntityResourceTest.testGetSpecWithNoExplicitAccepts:98->testGetSpec:126 expected [name: ent-with-spec
services:
- type: org.apache.brooklyn.entity.stock.BasicEntity] but found [[{"format":"brooklyn-camp","summary":"brooklyn-camp plan for null","contents":"name: ent-with-spec\nservices:\n- type: org.apache.brooklyn.entity.stock.BasicEntity"}]]

Tests run: 251, Failures: 2, Errors: 0, Skipped: 0
```

File: rest/rest-resources/src/test/java/org/apache/brooklyn/rest/resources/EntityResourceTest.java [hunk 2]
```diff
@@ -114,11 +114,11 @@ protected void testGetSpec(String acceptMimeType) throws Exception {
         // Retrieve the yaml spec, and confirm it is as expected (not wrapped in quotes, and treating \n sensibly)
         Response response;
         if (acceptMimeType != null) {
-            response = client().path("/applications/" + appName + "/entities/" + appName + "/spec")
+            response = client().path("/applications/" + appName + "/entities/" + appName + "/speclist")
```
Comment (ahgittin) [inline rest/rest-resources/src/test/java/org/apache/brooklyn/rest/resources/EntityResourceTest.java:117]: this is used in tests which expect the response from `/spec`; that's the test failures.  easy to fix of course!  i'll do that.

Comment (iuliana) [reply rest/rest-resources/src/test/java/org/apache/brooklyn/rest/resources/EntityResourceTest.java:117]: @ahgittin I might have alread done that and not pushed. I reemember running the build and noticed the test failure too. I did that temporary modification to check that my API was called.

**Comment by ahgittin** [comment]
yeah we've all done that!  simple fix, included in merge to master.  :)

**Pull request closed without merging**